src/prefect/cache_policies.py

```diff
@@ -1,17 +1,34 @@
 import inspect
+from copy import deepcopy
 from dataclasses import dataclass, field
-from typing import Any, Callable, Dict, List, Optional
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Union
+
+from typing_extensions import Literal
 
 from prefect.context import TaskRunContext
 from prefect.utilities.hashing import hash_objects
 
+if TYPE_CHECKING:
+    from prefect.filesystems import WritableFileSystem
+    from prefect.locking.protocol import LockManager
+    from prefect.transactions import IsolationLevel
+
 
 @dataclass
 class CachePolicy:
     """
     Base class for all cache policies.
     """
 
+    storage: Union["WritableFileSystem", str, Path, None] = None
+    isolation_level: Union[
+        Literal["READ_COMMITTED", "SERIALIZABLE"],
+        "IsolationLevel",
+        None,
+    ] = None
+    locks: Union[str, Path, "LockManager", None] = None
```

Comment:
this kwarg doesn't 100% read right to me, as `lock` is such a strong noun already. `lock_manager` might be better despite being longer? any other options?

Reply:
I called it `locks` because I was also allowing a file path as syntactic sugar for a `FileSystemLockManager`. I can try and think of an alternative name. Otherwise, I can remove that sugar, rename the kwarg to `lock_manager`, and accept just a `LockManager`.

Reply:
I updated this to `lock_manager` and removed the option to pass a path. Let me know what you think.